### PR TITLE
DXF export (2D) and exporters cleanup

### LINKED ISSUE #352: Export to DXF for 2D sections
In relation to #340 it might easier to implement export to DXF. It is a more common format used for CNC work.

#122 is somewhat related to this.

### PR DESCRIPTION
**Pull request by adam-urbanczyk**
Will resolve #352 , #340, #212, #334 and #360 .

This adds dxf export. Moreover I implemented a new export function and marked exportShape as deprecated:
```python
def export(
    w: Union[Shape, Workplane],
    fname: str,
    exportType: Optional[ExportLiterals] = None,
    tolerance: float = 0.1,
): ...
```
Now exporting is more user friendly: `exporters.export(obj,'out.step')`

### CODE REVIEW

File: doc/apireference.rst [hunk 1]
```diff
@@ -120,7 +120,7 @@ File Management and Export
 .. autosummary::
     importers.importStep
     importers.importDXF
-    exporters.exportShape
+    exporters.export
```
Comment (jmwright) [inline doc/apireference.rst:123]: I expect this change is going to break some existing scripts where users have used `exportShape` directly.

Comment (jmwright) [reply doc/apireference.rst:123]: I'm not saying "don't do it", I'm just pointing out that we may get issues filed over this change.

Comment (adam-urbanczyk) [reply doc/apireference.rst:123]: It is so far only marked as deprecated in the code. I'd propose to remove it in 2.2 so there is enough time for ppl to react on the deprecation warning.

Comment (jmwright) [reply doc/apireference.rst:123]: Sounds good, +1

**Pull request merged**